### PR TITLE
Initial attempt at conditions for kni-operator

### PR DESCRIPTION
**Pull request by djzager**

### CODE REVIEW

File: pkg/controller/knicluster/knicluster_controller.go [hunk 1]
```diff
@@ -276,11 +325,32 @@ func (r *ReconcileKNICluster) Reconcile(request reconcile.Request) (reconcile.Re
...
 		err = f(instance, reqLogger)
 		if err != nil {
+			status.SetStatusCondition(&instance.Status.Conditions, kniv1alpha1.Condition{
+				Type:    kniv1alpha1.ConditionDegraded,
+				Status:  corev1.ConditionTrue,
+				Reason:  "ReconcileFailed",
+				Message: fmt.Sprintf("Failed reconcilitation %v", err),
+			})
+
+			_ = r.client.Status().Update(context.TODO(), instance)
```
Comment (djzager) [inline pkg/controller/knicluster/knicluster_controller.go:335]: ignored the error here as we have already failed and don't want the err passed up to be about updating the status.

File: pkg/controller/knicluster/knicluster_controller.go [hunk 2]
```diff
@@ -251,6 +278,28 @@ func (r *ReconcileKNICluster) Reconcile(request reconcile.Request) (reconcile.Re
...
+		status.SetStatusCondition(&instance.Status.Conditions, kniv1alpha1.Condition{
+			Type:   kniv1alpha1.ConditionDegraded,
+			Status: corev1.ConditionFalse,
+		})
+		status.SetStatusCondition(&instance.Status.Conditions, kniv1alpha1.Condition{
+			Type:   kniv1alpha1.ConditionUpgradeable,
+			Status: corev1.ConditionUnknown,
+		})
+
+		_ = r.client.Status().Update(context.TODO(), instance)
```
Comment (djzager) [inline pkg/controller/knicluster/knicluster_controller.go:300]: I'm not totally sure what to do in the case of an err here. Do we fail the reconcile and requeue?

File: pkg/controller/knicluster/knicluster_controller.go [hunk 3]
```diff
@@ -276,11 +325,32 @@ func (r *ReconcileKNICluster) Reconcile(request reconcile.Request) (reconcile.Re
...
+				Message: fmt.Sprintf("Failed reconcilitation %v", err),
+			})
+
+			_ = r.client.Status().Update(context.TODO(), instance)
 			return reconcile.Result{}, err
 		}
 	}
 
-	return reconcile.Result{}, nil
+	status.SetStatusCondition(&instance.Status.Conditions, kniv1alpha1.Condition{
```
Comment (djzager) [inline pkg/controller/knicluster/knicluster_controller.go:340]: Probably need to dig deeper before marking `kni-operator` as "Available" but for now this illustrates the point.

**Comment by mhrivnak** [comment]
@djzager I assume we should close this one?

**Comment by djzager** [comment]
yup.

**Pull request closed without merging**